contracts/fuzz/scenarios/ChaosOps.sol

```diff
@@ -0,0 +1,909 @@
+            "Do not just mint RTokens/StRSR"
+        );
+        ERC20Fuzz(address(token)).mint(main.someUser(userID), amount);
+        require(token.totalSupply() <= 1e57, "Do not mint 'unreasonably' many tokens");
+    }
+
+    function burn(
+        uint8 userID,
+        uint8 tokenID,
+        uint256 amount
+    ) public {
+        IERC20 token = main.someToken(tokenID);
+        require(
+            address(token) != address(main.rToken()) && address(token) != address(main.stRSR()),
+            "Do not just burn RTokens/StRSR"
+        );
+        ERC20Fuzz(address(token)).burn(main.someUser(userID), amount);
+    }
+
+    // ==== user functions: asset registry ====
+
+    // refresh the state of all assets
+    function refreshAssets() public {
+        main.assetRegistry().refresh();
+    }
+
+    function registerAsset(
+        uint8 tokenID,
+        uint8 targetNameID,
+        uint256 defaultSeed,
+        uint256 delaySeed
+    ) public {
+        // TODO: Review - More randomness for all params? Pricemodels, TradingRange, etc?
+        IERC20 erc20 = main.someToken(tokenID);
+        IAssetRegistry reg = main.assetRegistry();
+        if (reg.isRegistered(erc20)) return;
+
+        string memory firstChar = getFirstChar(IERC20Metadata(address(erc20)).symbol());
+
+        if (strEqual(firstChar, "C") || strEqual(firstChar, "S")) {
+            CollateralMock newColl = new CollateralMock(
+                IERC20Metadata(address(erc20)),
+                IERC20Metadata(address(0)), // no reward
+                defaultParams().rTokenTradingRange,
+                uint192(between(1, 1e18, defaultSeed)), // def threshold
+                between(1, type(uint256).max, delaySeed), // delay until default
+                IERC20Metadata(address(0)),
+                someTargetName(targetNameID),
+                justOne,
+                stable,
+                justOne,
+                justOne
+            );
+            reg.register(newColl);
+        } else {
+            AssetMock newAsset = new AssetMock(
+                IERC20Metadata(address(erc20)),
+                IERC20Metadata(address(0)), // no recursive reward
+                defaultParams().rTokenTradingRange,
+                volatile
+            );
+            reg.register(newAsset);
+        }
+    }
+
+    function swapRegisteredAsset(
+        uint8 tokenID,
+        uint256 defaultSeed,
+        uint256 delaySeed,
+        uint256 switchTypeSeed
+    ) public {
+        // TODO: Review - More randomness for all params? Pricemodels, TradingRange, etc?
+        IERC20 erc20 = main.someToken(tokenID);
+        IAssetRegistry reg = main.assetRegistry();
+        if (!reg.isRegistered(erc20)) return;
+
+        IAsset asset = reg.toAsset(erc20);
+
+        // Switch type Asset -> Coll and viceversa one out of 100 times
+        bool switchType = (switchTypeSeed %= 100) <= 1;
```

Comment:
This will make switchType true 2 out of 100 times -- which is fine, but violates the stated intent.

Reply:
Also, using the `%=` operator here is slightly funky; why not just `%`?